tests/grids/test_refinement.py

```diff
@@ -38,25 +40,88 @@ def test_grid_perturbation_1d_internal_nodes_fixed(self):
 
 
 class TestGridRefinement1D:
-    @pytest.mark.parametrize(
-        "g,target_nodes",
-        [
-            (TensorGrid(np.array([0, 2, 4])), np.arange(5)),  # uniform
-            (TensorGrid(np.array([0, 2, 6])), np.array([0, 1, 2, 4, 6])),  # non-uniform
-        ],
-    )
-    def test_refinement_grid_1D(self, g, target_nodes):
-        h = refinement.refine_grid_1d(g, ratio=2)
-        assert np.allclose(h.nodes[0], target_nodes)
-
-    def test_refinement_grid_1d_general_orientation(self):
-        x = np.array([0, 2, 6]) * np.ones((3, 1))
-        g = TensorGrid(x[0])
-        g.nodes = x
-        h = refinement.refine_grid_1d(g, ratio=2)
-        assert np.allclose(
-            h.nodes, np.array([[0, 1, 2, 4, 6], [0, 1, 2, 4, 6], [0, 1, 2, 4, 6]])
+    """Tests for the 1D grid refinement functions.
+
+    The test focuses on the topology of the refined grid, also in cases with non-uniform
+    grid spacing, permuted node/cell orderings and split nodes.
+
+    Geometric aspects (whatever that would be, perhaps grids not aligned with the
+    x-axis) are not tested, as the chance of errors in the geometric computations is
+    considered low.
+
+    """
+
+    def compare_refined_grids(self, g: pp.Grid, g_ref: pp.Grid, ratio: int) -> None:
+        """Helper function to compare the refined grid with the original grid.
+
+        See comments in the below code for which properties are compared.
+
+        Parameters:
+            g: The original grid.
+            g_ref: The refined grid.
+            ratio: The ratio of the number of cells in the refined grid to the original
+                grid.
+
+        """
+
+        # The two grids should cover the same domain.
+        assert np.sum(g.cell_volumes) == np.sum(g_ref.cell_volumes)
+
+        # If the ratio is given, all cells should be refined by this ratio.
+        if ratio is not None:
+            assert np.allclose(g_ref.num_cells / g.num_cells, ratio)
+
+        # There should be as many faces as nodes in a 1d grid. Check that the face-node
+        # map is 1-1, and that there are as many node coordinates as there are nodes.
```

Comment:
The test for 1-1 tests only necessary but not sufficient conditions. As the grids tested here are small, checking whether `np.abs(np.linalg.det(g_ref_face_nodes))==1` should be feasible and do the job.

Reply:
Agree, will implement.

Reply:
Implemented.